mirrord-layer/src/lib.rs

```diff
@@ -386,33 +364,56 @@ fn enable_hooks(enabled_file_ops: bool, enabled_remote_dns: bool) {
     interceptor.end_transaction();
 }
 
+// TODO: When this is annotated with `hook_guard_fn`, then the outgoing sockets never call it (we
+// just bypass). Everything works, so, should we intervene?
+//
```

Comment:
As far as I follow (AFAIF?) close will close our "proxy" socket, which will then `layer` will see it in the `interceptor_task` and then will notify the remote agent. sounds good. maybe write this flow?

Reply:
Added a flow for closing it on `agent`.